### PR TITLE
feat: Use ERIC identity instead of applicantDetails.emailAddress

### PR DESCRIPTION
**Pull request by rhysbarrett**
### JIRA

[Remove "Email" field from "Applicant details" screen](https://companieshouse.atlassian.net/browse/SR-124) 

### Change description

Support removal of Applicant Details > Email Address field from Front-end by: 
- Removing `emailAddress` field from `ApplicantDetails`
- Adding `createdBy` field to root-level `Suppression` object
- Obtaining `createdBy` value from ERIC `ERIC-Authorised-User` header instead of POST data from the frontend.
- Using `createdBy` value as address for sending Confirmation email by first copying value to `ApplicationReceivedEmailData` object under attribute name `contactEmailAddress`

### Related PRs

- companieshouse/suppression-web#132
- companieshouse/chs-notification-api#234

### Work Checklist

- [x] Tests added where applicable
- [x] Test coverage of new code meets target of 80%

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/controller/SuppressionController.java [hunk 1]
```diff
@@ -59,16 +59,16 @@ public SuppressionController(SuppressionService suppressionService, Logger logge
...
 
         if (StringUtils.isBlank(userId)) {
             return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
         }
 
         try {
-
-            final String id = suppressionService.saveSuppression(applicantDetails);
-
+            final String createdBy = authorisedUser.split(";")[0];
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/controller/SuppressionController.java:70]: can `authorisedUser` be `null`/`empty`? It may be worth adding to the `isBlank` check above and returning 401

File: src/main/java/uk/gov/companieshouse/model/Suppression.java [hunk 2]
```diff
@@ -13,6 +13,8 @@
     @JsonIgnore
     private LocalDateTime createdAt;
 
+    private String createdBy;
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/model/Suppression.java:16]: the `Suppression` data is returned by the `getSuppression` endpoint, do we need to expose the email address to the client?

File: src/test/java/uk/gov/companieshouse/controller/SuppressionControllerTest_POST.java [hunk 3]
```diff
@@ -21,12 +21,14 @@
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
```
Comment (n-r-b) [inline src/test/java/uk/gov/companieshouse/controller/SuppressionControllerTest_POST.java:21]: please add a test covering an empty/invalid authorised user header

File: src/test/java/uk/gov/companieshouse/mapper/SuppressionMapperTest.java [hunk 4]
```diff
@@ -61,8 +61,8 @@ void shouldReturnNullWhenValueIsNull() {
         @Test
         void shouldMapValueWhenValueIsNotNull() {
 
-            SuppressionEntity mapped = mapper.map(new Suppression(null, applicationReference,
-                new ApplicantDetails(fullName, previousName, emailAddress, dateOfBirth),
+            SuppressionEntity mapped = mapper.map(new Suppression(null, "", applicationReference,
```
Comment (n-r-b) [inline src/test/java/uk/gov/companieshouse/mapper/SuppressionMapperTest.java:64]: can we reference the `TestData.Suppression.createdBy` here instead of using an empty string

Comment (n-r-b) [reply src/test/java/uk/gov/companieshouse/mapper/SuppressionMapperTest.java:64]: we're also missing an assertion for the mapping of `createdBy`

File: src/test/java/uk/gov/companieshouse/mapper/SuppressionMapperTest.java [hunk 5]
```diff
@@ -115,8 +114,8 @@ void shouldReturnNullWhenValueIsNull() {
 
         @Test
         void shouldMapValueWhenValueIsNotNull() {
-            Suppression mapped = mapper.map(new SuppressionEntity(applicationReference, createdAt,
-                new ApplicantDetailsEntity(fullName, previousName, emailAddress, dateOfBirth),
+            Suppression mapped = mapper.map(new SuppressionEntity(applicationReference, createdAt, createdBy,
```
Comment (n-r-b) [inline src/test/java/uk/gov/companieshouse/mapper/SuppressionMapperTest.java:117]: we're missing an assertion here too for the mapped `createdBy`

File: src/main/java/uk/gov/companieshouse/controller/SuppressionController.java (hunk 1 above)
Comment (rhysbarrett) [reply src/main/java/uk/gov/companieshouse/controller/SuppressionController.java:70]: Ah, good idea!

File: src/main/java/uk/gov/companieshouse/model/Suppression.java (hunk 2 above)
Comment (rhysbarrett) [reply src/main/java/uk/gov/companieshouse/model/Suppression.java:16]: It's the easiest way to pass the value to the NotificationAPI - I tried using @JsonIgnore on it at first but realised that it causes the value to be missing from the email data. 

I could easily get around this by manually adding it to the ApplicationReceivedEmailData but it feels like an unnecessary extra step if the createdBy field isn't sensitive enough to warrant masking it.

What do you reckon? Is it sensitive?

File: src/test/java/uk/gov/companieshouse/controller/SuppressionControllerTest_POST.java (hunk 3 above)
Comment (rhysbarrett) [reply src/test/java/uk/gov/companieshouse/controller/SuppressionControllerTest_POST.java:21]: Ah, yes, sure.

File: src/main/java/uk/gov/companieshouse/model/Suppression.java (hunk 2 above)
Comment (n-r-b) [reply src/main/java/uk/gov/companieshouse/model/Suppression.java:16]: ok, i see what you mean. I would say the email address is sensitive data (hence the reason for this ticket), the `getSuppression` endpoint already returns the applicant details and address. If it's not too much work, i suggest we mask this

Comment (rhysbarrett) [reply src/main/java/uk/gov/companieshouse/model/Suppression.java:17]: Bart delegated this question / decision to Damian, who agreed that it probably isn't necessary to mask the email address. However there's no downside to masking it, so I've made the necessary changes to support this and updated the Swagger screenshot to reflect the JSON schema change.

Comment (n-r-b) [reply src/main/java/uk/gov/companieshouse/model/Suppression.java:17]: thanks for this!

**Review by n-r-b — APPROVED** [review]
